### PR TITLE
Adding ability to configure resources for weave (#8113)

### PR DESCRIPTION
**Pull request by mmerrill3**
This is a PR to address #8113 whereby kops users can gain the ability to configure resources for the weave and weave-npc containers inside the weave daemonset.
Signed-off-by: mmerrill3 <michael.merrill@vonage.com>

### CODE REVIEW

**Comment by mikesplain** [comment]
Thanks @mmerrill3!

/lgtm
/approve

Fixes: #8113

**Comment by hakman** [comment]
@mmerrill3 @mikesplain @rifelpet Any chance this broke the weave plugin? My gu
https://testgrid.k8s.io/sig-cluster-lifecycle-kops#kops-aws-weave
https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-kops-aws-weave/1214998660220194818/artifacts/ip-172-20-50-49.sa-east-1.compute.internal/protokube.log
```
Error from server (BadRequest): error when creating "/tmp/channel468225765/manifest.yaml": DaemonSet in version "v1" cannot be handled as a DaemonSet: v1.DaemonSet.Spec: v1.DaemonSetSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Resources: v1.ResourceRequirements.Requests: Limits: unmarshalerDecoder: unable to parse quantity's suffix, error found in #10 byte of ...|y":"200mi"},"request|..., bigger context ...|rt":6784}},"resources":{"limits":{"memory":"200mi"},"requests":{"cpu":"50m","memory":"200mi"}},"secu|...
Error: error updating "networking.weave": error applying update from "s3://k8s-kops-prow/e2e-kops-aws-weave.test-cncf-aws.k8s.io/addons/networking.weave/k8s-1.12.yaml": error running kubectl
```

File: upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -179,10 +179,13 @@ spec:
               port: 6784
           resources:
             requests:
-              cpu: 50m
-              memory: 200Mi
+              cpu: {{ or .Networking.Weave.CPURequest "50m" }}
+              memory: {{ or .Networking.Weave.MemoryRequest "200mi" }}
```
Comment (rifelpet) [inline upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: These memory units need to have a capital `M` don't they?

https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource/#memory-units

Comment (hakman) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: That's what I thought also...

**Pull request merged**